### PR TITLE
Add OverTheSpot support.

### PR DESCRIPTION
**Pull request by odknt**
Changes proposed in this pull request:

* Implement OverTheSpot.
* Add `imstyle` option. (defaults taken `root`)
* Increase buffer size for preedit area. 

imstyle_root imstyle_overthespot

### CODE REVIEW

**Comment by neeasade** [comment]
TIL of OverTheSpot. In that gif are you using https://linux.die.net/man/1/kinput2 ?

**Comment by odknt** [comment]
I used UIM in the gif.
This PR is add `OverTheSpot` by using XIM, so that we can use many input methods.

**Comment by actionless** [comment]
do we really need a `goto` here?

**Comment by odknt** [comment]
I think unnecessary too, so I fixed it.

File: src/xst.c [hunk 1]
```diff
@@ -542,6 +545,9 @@ static void *xmalloc(size_t);
 static void *xrealloc(void *, size_t);
 static char *xstrdup(char *);
 
+static int isavailablestyle(XIMStyle);
```
Comment (actionless) [inline src/xst.c]: what about isavailable**im**style?

Comment (odknt) [reply src/xst.c]: Thank you, I'll do rename the function.

**Pull request merged**